src/components/MapController/MapController.js

```diff
@@ -52,25 +84,42 @@ var MapController = React.createClass({
                     ]
     var color_scales = [['false', 'Linear'], ['true', 'Logarithmic']]
 
+    var ids = this.props.meta.map(function(el){return el.unique_id})
+
+    // Only render map if required paramers exist in state/props
```

Comment:
I don't like this. If the user is on a slower network, or there's a lag on the metadata AJAX request, then there's a giant blank white space on the page until it returns. Plus then it delays the start of the WMS tile loading until after the metadata comes back. Basically, it makes everything wait on one single request, even things that shouldn't be dependent on it.

Can't we just render the map with time, dataset and variable undefined?